src/ts/app/view/viewport/handler/mouse.ts

```diff
@@ -117,6 +127,111 @@ export class MouseHandler {
         listenOnce(document, 'mouseup', this.shiftOnMouseUp)
     }
 
+    selectionHandlePressed = () => {
+        console.log('selection handle pressed!')
+        // before anything else, disable the camera
+        this.viewport.cameraIsLocked = true
+
+        // record the starting positions of selected landmarks
+        this.dragStartPositions = this.viewport.selectedLandmarks
+            .map(lm => [lm.index, lm.point.clone()])
+
+        // start listening for dragging selection handle
+        document.addEventListener('mousemove', this.selectionHandleOnDrag)
+        listenOnce(document, 'mouseup', this.selectionHandleOnMouseUp)
+    }
+
+    selectionBoxPressed = (event: MouseEvent) => {
+        console.log('selection box pressed!')
+        // before anything else, disable the camera
+        this.viewport.cameraIsLocked = true
+
+        // record the starting positions of selected landmarks
+        this.positionLmDrag.copy(new THREE.Vector2(event.clientX, event.clientY))
+        this.dragStartPositions = this.viewport.selectedLandmarks
+            .map(lm => [lm.index, lm.point.clone()])
+
+        // start listening for dragging selection box
+        document.addEventListener('mousemove', this.landmarkOrSelectionBoxOnDrag)
+        listenOnce(document, 'mouseup', this.selectionBoxOnMouseUp)
+    }
+
+    rotationHandlePressed = () => {
+        console.log('rotation handle pressed!')
+        // before anything else, disable the camera
+        this.viewport.cameraIsLocked = true
+        this.viewport.activateRotationCircle()
+
+        this.viewport.clearCanvas()
+        this.viewport.drawRotationCircle()
+
+        // record the starting positions of selected landmarks
+        this.dragStartPositions = this.viewport.selectedLandmarks
+            .map(lm => [lm.index, lm.point.clone()])
+
+        // start listening for dragging rotation handle
+        document.addEventListener('mousemove', this.rotationHandleOnDrag)
+        listenOnce(document, 'mouseup', this.rotationHandleOnMouseUp)
+    }
+
+    mouseDownOnSelectionHandle = () => {
+        const md = this.onMouseDownPosition
+        const sb = this.viewport.selectionBox
+        const hr = sb.handleRadius
+        if (sb.minPosition.x === -1 && sb.minPosition.y === -1 && sb.maxPosition.x === -1 && sb.maxPosition.y === -1) {
+            console.log("selection box inactive!")
+            return false
+        }
+        // add one pixel padding to make handle easier to grab
+        if (md.x <= sb.minPosition.x + hr + 1 && md.x >= sb.minPosition.x - hr - 1
+        && md.y <= sb.minPosition.y + hr + 1 && md.y >= sb.minPosition.y - hr - 1) {
```

Comment:
let's create a helper function at the top of `mouse.ts`:
```typescript
function boxIntersection(mousePosition, boxCenter, radius) {
    return (mousePosition.x <= boxCenter.x + radius && 
            mousePosition.x >= boxCenter.x - radius &&
            mousePosition.y <= boxCenter.y + radius &&
            mousePosition.y >= boxCenter.y - radius)
}
```
(Add the appropriate types ofc)

Then call as:

```typescript
// Add small extra padding to mouse intersection logic
const hr = sb.handleRadius + RESIZE_HANDLE_PADDING
```
```typescript
if (boxIntersection(md, sb.minPosition, hr)) {
...
```
Repeat pattern for the circular intersection, so top of file we can have


```typescript
// Small extra padding so users don't have to be perfect with mouse targeting
const RESIZE_HANDLE_PADDING = 1
const ROTATION_HANDLE_PADDING = 2

function boxIntersection(mousePosition, boxCenter, radius) {
...
}

function circleIntersection(mousePosition, circleCenter, radius) {
...
}
```